lib/modules/datasource/maven/util.ts

```diff
@@ -91,13 +102,78 @@ export async function downloadHttpProtocol(
       // istanbul ignore next
       logger.debug({ failedUrl }, 'Unsupported host');
     } else {
-      logger.info({ failedUrl, err }, 'Unknown error');
+      logger.info({ failedUrl, err }, 'Unknown HTTP download error');
     }
     return {};
   }
 }
 
-export async function checkHttpResource(
+function isS3CedentialsError(err: { name: string; message: string }): boolean {
+  return err.name === 'CredentialsProviderError';
+}
+
+function isS3RegionError(err: { name: string; message: string }): boolean {
+  return err.message === 'Region is missing';
+}
+
+function isS3NotFound(err: { name: string; message: string }): boolean {
+  return err.message === 'NotFound' || err.message === 'NoSuchKey';
+}
+
+function parseS3Url(rawUrl: string): { Bucket: string; Key: string } {
+  const parsedUrl = parseUrl(rawUrl) as url.URL;
```

Comment:
```suggestion
  const parsedUrl = parseUrl(rawUrl);
```

nope, totally wrong. don't use node `url` module for new code, as it's deprecated.